website/docs/plugins/provider.html.md

```diff
@@ -32,6 +32,44 @@ of this page will assume you're familiar with
 [plugin basics](/docs/plugins/basics.html) and that you already have
 a basic development environment setup.
 
+## Provider Plugin Codebases
+
+Provider plugins live outside of the Terraform core codebase in their own
+source code repositories. The official set of provider plugins released by
+Hashicorp (developed by both Hashicorp staff and community contributors)
+all live in repositories in
+[the `terraform-providers` organization](https://github.com/terraform-providers)
+on GitHub, but third-party plugins can be maintained in any source code
+repository.
+
+When developing a provider plugin, it is recommended to use a common `GOPATH`
+that includes both the core Terraform repository and the repositories of any
+providers being changed. This makes it easier to use a locally-built
+`terraform` executable and a set of locally-built provider plugins together
+without further configuration. For example, for the `template` provider:
+
+```
+$ go get github.com/hashicorp/terraform
+...
+$ go get github.com/terraform-providers/terraform-provider-template
+...
+$ go install github.com/hashicorp/terraform
```

Comment:
True enough. I was trying to show that you can use `install` to build separately from downloading, but combining this all together into one usage obscures that. I'll revise it so that it's given as two separate examples: how to install, and then how to build.

You're right that most readers will know how to use the `go` subcommands, but I mainly just wanted to draw attention to the fact that these two different packages are _both_ required for a working dev environment. When I revise I'll also try to write something short to help clarify what sorts of changes require a compile of core vs. the provider and how to update the vendoring of Terraform core into the provider, since I know some people were already somewhat confused about these things.